data/lab: Add quizzes (#203)
open-education-hub/operating-systems: content/chapters/data/lab/quiz/malloc-mmap.md

# Malloc `mmap()`

## Question Text

When does `malloc()` use `mmap()`?

## Question Answers

> - When you want read-only memory
> 
> - When you want zeroed memory
> 
> + When you allocate chunks of memory bigger than an internal threshold

Reviewer: ```suggestion
- When it allocates read-only memory

- When it allocates zeroed memory

+ When it allocates chunks of memory bigger than an internal threshold
```

- When the heap is full

## Feedback

> `malloc` uses both `brk` and `mmap`, but preffers `mmap` for big chunks of memory (bigger than 128 KB).

Reviewer: ```suggestion
`malloc` uses both `brk()` and `mmap()`, but prefers `mmap()` for big chunks of memory (by default larger than 128 KB).
This value can be altered using [`mallopt()`](https://linux.die.net/man/3/mallopt) with the `param` argument set to `M_MMAP_THRESHOLD`.
```
